src/Microsoft.AspNet.Mvc.Xml/RequiredValidationHelper.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Reflection;
+using System.Runtime.Serialization;
+using Microsoft.AspNet.Mvc.ModelBinding;
+
+namespace Microsoft.AspNet.Mvc.Xml
+{
+    /// <summary>
+    /// Validates types having value type properties decorated with <see cref="RequiredAttribute"/>
+    /// but no <see cref="DataMemberAttribute"/>.
+    /// </summary>
+    public static class RequiredValidationHelper
+    {
+        private static ConcurrentDictionary<Type, Dictionary<Type, List<string>>> cachedValidationErrors
```

Comment:
Discussed this offline (This could be made non static and the whole type to be non-static and the formatter could be instantiating this type, but the issue is that formatters even though are registered typically single instance, they can be DI injectable and be created on per-request basis too in which case the presence of cache is useless.)